vertx/src/main/java/com/groupon/SumVerticle.java

```diff
@@ -0,0 +1,64 @@
+package com.groupon;
+
+import java.util.List;
+
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.rxjava.core.AbstractVerticle;
+import io.vertx.rxjava.core.Vertx;
+import io.vertx.rxjava.core.eventbus.Message;
+import io.vertx.rxjava.core.http.HttpClient;
+import io.vertx.rxjava.core.http.HttpClientRequest;
+import io.vertx.rxjava.core.http.HttpClientResponse;
+import rx.Observable;
+
+import com.groupon.common.Method;
+import com.groupon.common.Utils;
+
+public class SumVerticle extends AbstractVerticle {
+
+    private HttpClient httpClient;
+
+    public SumVerticle(Vertx vertx) {
+        httpClient = vertx.createHttpClient(
+                new HttpClientOptions()
+                        .setKeepAlive(false)
+                        .setMaxPoolSize(64) // 5 default // this is per host
+                        .setLogActivity(false)
+                        .setMaxWaitQueueSize(-1) // -1 default
+                        //.setPipelining(true)
+                        //.setPipeliningLimit(100)
+                        .setProtocolVersion(HttpVersion.HTTP_1_0) // 1.1 default
+                        .setConnectTimeout(1_000_000)
+                        .setIdleTimeout(1_000) // For some reasons now connections are established
+                        // with 0.
+                        .setTcpNoDelay(true) // Same as in jetty.
+                        .setTryUseCompression(false));
+    }
+
+    @Override
+    public void start() throws Exception {
+        Observable<Message<List<Integer>>> sumMemory = vertx.eventBus().<List<Integer>>localConsumer(Method.METHOD_EVENTBUS_NETWORK).toObservable();
+        sumMemory.subscribe(message -> callSumNetwork(message.body())
+                .subscribe(body -> {
+                    int sum = Integer.valueOf(body.substring(0, body.indexOf('\n')));
+                    // logger.debug("Received from network: " + sum);
+                    message.reply(sum);
+                }, fail -> message.fail(500, fail.getMessage())));
+    }
+
+    public Observable<String> callSumNetwork(List<Integer> values) {
```

Comment:
This is also used as a util method from MainVerticle. Maybe move it to a different class (directly invoking between verticles is weird)?

Reply:
removed the verticle and turned it into a utility class instead